core/src/epicli/data/common/ansible/playbooks/roles/repository/files/download-requirements/centos-7/download-requirements.sh

```diff
@@ -79,30 +79,40 @@ create_directory() {
 disable_repo() {
 	local repo_id="$1"
 
-	if yum repolist enabled | grep --quiet "$repo_id"; then
+	if [[ $(yum repolist enabled | grep --quiet "$repo_id") ]]; then
 		echol "Disabling repository: $repo_id"
 		yum-config-manager --disable "$repo_id" ||
 			exit_with_error "Command failed: yum-config-manager --disable \"$repo_id\""
 	fi
 }
 
-# params: <file_url> <dest_dir>
+# params: <file_url> <dest_dir> <new_filename> # $3 not required
 download_file() {
 	local file_url="$1"
 	local dest_dir="$2"
 
-	local file_name=$(basename "$file_url")
-	local dest_path="$dest_dir/$file_name"
-	local retries=3
-
-	# wget with --timestamping sometimes failes on AWS with ERROR 403: Forbidden
-	# so we remove existing file to overwrite it, to be optimized
-	[[ ! -f $dest_path ]] || remove_file "$dest_path"
+	if [[ ${3-} ]]; then
+		local file_name=$3
+	else
+		local file_name=$(basename "$file_url")
+	fi
 
-	echol "Downloading file: $file_url"
+	local dest_path="${dest_dir}/${file_name}"
+	local retries=3
 
-	run_cmd_with_retries wget --no-verbose --directory-prefix="$dest_dir" "$file_url" $retries ||
-		exit_with_error "Command failed: wget --no-verbose --directory-prefix=\"$dest_dir\" \"$file_url\""
+	if [[ -f $dest_path ]]; then
+		echo "$file_name already exists. Skipping..."
```

Comment:
Then, probably deployment will fail in case someone will use the broken file. In case of grafana dashboards it not so relevant but in case of zip's can be. The condition gives us faster download-requirements execution but as you point right, can skip broken file. So would be good to have here some non-time consuming mechanism to validate if file is downloaded completely. Maybe prepare .md5 of files and compare (maybe this can be done through Ansible?) but again, its case for refactoring.